tests/PHPStan/Analyser/data/ext-ds.php

```diff
@@ -16,6 +16,12 @@ class B
 
 class Foo
 {
+	/** @param Map<int, int|string> $a */
+	public function mapGetUnionType(Map $a) : void
+	{
+		assertType('int|string|TDefault (method Ds\Map::get(), parameter)', $a->get(1));
```

Comment:
This isn't ideal. The result should be `int|string|null`. We should resolve TDefault to the default parameter value.

That's logic that should be done here https://github.com/phpstan/phpstan-src/blob/9b712a90e3fa03a0d26b51ec2ff35c807ab06d11/src/Reflection/GenericParametersAcceptorResolver.php#L33 but isn't. Can you look into it? My theory is that ext-ds reflection gives us wrong data.

Reply:
We should achieve 
`assertType('int|string|', $a->get(1));`  
`assertType('int|string|null', $a->get(1, null));`

Can't do that with just template annotations I believe. I'll try to edit the extension.

Reply:
What happens when `$a->get(1)` is not found?